File: 06Adapter/Students/2019/SebastianPiekarek/src/ZAD1/Experiment.java
package ZAD1;

public class Experiment {

    public static void main(String[] args) {
        System.out.println("Experiment 1: test the aircraft engine");
        IAircraft aircraft = new Aircraft();
        aircraft.takeOff();
        if (((Aircraft) aircraft).getAirborne()) System.out.println("The aircraft engine is fine, flying at " +
        ((Aircraft) aircraft).getHeight() + "meters");

        System.out.println("\nExperiment 2: Use the engine in the Seabird");
        IAircraft seabird = new Seabird();
        seabird.takeOff();
        System.out.println("The Seabird took off");


        System.out.println("\nExperiment 3: Increase the speed of the Seabird:");
        ((Seabird) seabird).IncreaseRevs();
        ((Seabird) seabird).IncreaseRevs();
        if(((Seabird) seabird).getAirbone()){
            System.out.println("Seabird flying at height" + ((Seabird) seabird).getHeight() +
            " meters and speed " + ((Seabird) seabird).getSpeed() + " knots");
            System.out.println("Experiments successful; the Seabird flies!");
        }


        System.out.println("\nExperiment 4: Seabird as Aircraft object:");
        Aircraft seabirdTwo = new SeabirdTwo();
        seabirdTwo.takeOff();
        System.out.println("The Seabird took off");
        ((SeabirdTwo) seabirdTwo).IncreaseRevs();
        ((SeabirdTwo) seabirdTwo).IncreaseRevs();
        if( seabirdTwo.getAirborne()){
            System.out.println("Seabird flying at height" + seabirdTwo.getHeight() +
                    " meters and speed " + ((SeabirdTwo) seabirdTwo).getSpeed() + " knots");
            System.out.println("Experiments successful; the Seabird flies!");
        }





    }
}

File: 06Adapter/Students/2019/KawaPiotr/src/main/kotlin/pkawa/zad1/Experiment.kt
package pkawa.zad1

fun main() {
    // No adapter
    println("Experiment no. 1: test the aircraft engine")
    val aircraft: IAircraft = Aircraft()
    aircraft.takeOff()
    if (aircraft.airborne) println("The aircraft engine is fine, flying at ${aircraft.height} meters")

    // Classic usage of an adapter
    println("\nExperiment 2: Use the engine in the Seabird")
    val seabird: IAircraft = Seabird()
    seabird.takeOff() // And automatically increases speed
    println("The Seabird took off")

    // Two-way adapter: using seacraft instructions on an IAircraft object
    // (where they are not in the IAircraft interface)
    println("\nExperiment 3: Increase the speed of the Seabird:")
    (seabird as ISeacraft).increaseRevs()
    (seabird as ISeacraft).increaseRevs()
    if (seabird.airborne) println("Seabird flying at height ${seabird.height} meters and speed ${seabird.speed} knots")
    println("Experiments successful; the Seabird flies!")


    println("\nExperiment 4: Seabird NextGen as an Aircraft object:")
    val seabirdNextGen: Aircraft = SeabirdNextGen()
    seabirdNextGen.takeOff()
    println("The Seabird NextGen took off")
    (seabirdNextGen as SeabirdNextGen).increaseRevs()
    seabirdNextGen.increaseRevs()

    if (seabird.airborne) println("Seabird NextGen flying at height ${seabird.height} meters and speed ${seabird.speed} knots")
    println("Experiments successful; the Seabird flies!")
}